alf/algorithms/mi_estimator_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import unittest
+
+from absl import logging
+from absl.testing import parameterized
+import tensorflow as tf
+
+from alf.algorithms.mi_estimator import MIEstimator, ScalarAdaptiveAverager
+
+
+class MINEstimatorTest(parameterized.TestCase, unittest.TestCase):
+    @parameterized.parameters(
+        dict(estimator='DV', rho=0.0, eps=0.02),
+        dict(estimator='KLD', rho=0.0, eps=0.02),
+        dict(estimator='JSD', rho=0.0, eps=0.02),
+        dict(estimator='DV', rho=0.5, eps=0.3),
+        dict(estimator='DV', rho=0.5, eps=0.3, sampler='double_buffer'),
+        dict(estimator='DV', rho=0.5, eps=0.3, buffer_size=1048576),
+        dict(estimator='DV', rho=0.5, eps=0.6, sampler='shuffle'),
+        dict(estimator='DV', rho=0.5, eps=0.3, sampler='shift'),
+        dict(estimator='KLD', rho=0.5, eps=0.3),
+        dict(estimator='JSD', rho=0.5, eps=0.3),
+        dict(estimator='DV', rho=0.9, eps=7.0),
+        dict(estimator='KLD', rho=0.9, eps=7.0),
+        dict(estimator='JSD', rho=0.9, eps=12.0),
+    )
+    def test_mi_estimator(self,
+                          estimator='DV',
+                          sampler='buffer',
+                          rho=0.9,
+                          eps=1000.0,
+                          buffer_size=65536,
+                          dim=20):
+        mi_estimator = MIEstimator(
+            x_spec=tf.TensorSpec(shape=(dim, ), dtype=tf.float32),
+            y_spec=tf.TensorSpec(shape=(dim, ), dtype=tf.float32),
+            fc_layers=(512, ),
```

Comment:
Is it possible that fc_layers are not enough complex for JSD so that the converged T doesn't provide good estimates for log(P/Q)?